app/forms/openid_connect_authorize_form.rb

```diff
@@ -219,21 +203,21 @@ def error_redirect_uri
   end
 
   def scopes
-    if ialmax_requested? || ial2_requested? || ial2_strict_requested?
+    if ial_context.ialmax_requested? || ial2_or_greater?
       return OpenidConnectAttributeScoper::VALID_SCOPES
     end
     OpenidConnectAttributeScoper::VALID_IAL1_SCOPES
   end
 
   def validate_privileges
-    if (ial2_requested? && !ial2_service_provider?) ||
-       (ialmax_requested? && !ial2_service_provider?)
+    if (ial_context.ial2_requested? && !ial_context.ial2_service_provider?) ||
+       (ial_context.ialmax_requested? && !ial_context.ial2_service_provider?)
       errors.add(:acr_values, t('openid_connect.authorization.errors.no_auth'))
     end
   end
 
   def validate_liveness_checking_enabled_if_ial2_strict_requested
-    return unless ial2_strict_requested? && !FeatureManagement.liveness_checking_enabled?
+    return if !ial_context.ial2_strict_requested? || FeatureManagement.liveness_checking_enabled?
```

Comment:
flipped from `unless` to `if` mostly because of the line length limit being hit, but also I find `if` clearer to read for compound statements